<html>
<head>
<title>GHMM: General Hidden Markov Model library: HMMEd</title>
</head>
<style type="text/css" media="all">
@import "layout1.css" screen;
@import "print.css" print;
</style>
<div id="Content">



<table cellpadding="10" >
<tr><td><img src="GQL.gif"></td><td valign="bottom">
<h3>The Graphical Query Language:<br>
A GHMM-based tool for querying and clustering 
Gene-Expression time-course data</h3><p>

<em>(c) 2003-2004 Alexander Schliep
(c) 2004-2011 Ivan G. Costa and Alexander Schliep
</em>
</td>
</tr>
</table>

<p>GQL is a suite of tools for analyizing time-course experiments.
Currently, it is adapted to gene expression data. The two main tools
are GQLQuery, for querying data sets, and GQLCluster, which provides a
way for computing groupings based on a number of methods (model-based
clustering using HMMs as cluster models and estimation of a mixture of
HMMs). 



<h4>Availability</h4>

<p>GQL is freely available under the GPL. The first public release of
GQL is available in source form and binaries form Windows or Mac.</p>

<!--<center>
Download version 1.0
<A HREF="http://algorithmics.molgen.mpg.de/Downloads/GQL-Unix-1.0.zip">
GQL-Unix-1.0.zip</A>.
</center> -->

<p>Note that downloading and using GQL implies acceptance of the GPL.
GQL is not freeware, nor public domain and the copyright will be
enforced. As the GPL has strong consequences for any work derived from
GQL, commercial entities can inquire about non-exclusive licenses.</p>

<p>If you use GQL in your research, please do cite the aforementioned
paper(s).</p>


<h4>GQLQuery: Querying time-courses</h4>

<p>The GUI has been ported to Python using Tkinter and the brand-new
Python bindings for GHMM. It runs on all Linux/Unix boxes. Executable binaries for
MAC and Windows are provided.</p>

<center>
<img src="GQL-screen.gif">
</center>

<h4>GQLCluster: Finding groups in time-courses</h4>


<center>
<img src="gql/GQLCluster.gif">
</center>


<H2>Linux & Unix Version</H2> 

<H4>Prerequisites</H4> 

<p>You will need to install the following packages before you
install <A HREF="http://ghmm.org">GHMM</A>. The version 1.0 of GQL 
will only work with the most recent version of GHMM. 
<ul>
<li> Tk (as in Tcl/Tk) in a version newer than 8.3.x. There are binary
packages for most Linux distributions. You can find sources at <A
href="http://www.tcl.tk/software/tcltk/">www.tcl.tk/software/tcltk/</a>.
Tk provides the GUI toolkit.

<li> Python version 2.6.x. This is the interpreter for language
Python, the GQL-software is written in. Again, most Linux
distributions have binary packages for Python. Make sure to choose
Tkinter (the link to the Tk-framework) support. Sources etc.
can be found at <A HREF="http://www.python.org">www.python.org</A>.
Note that GQL does not run with version 3.X of Python.

<li> GSL, the GNU scientific library is at <A
HREF="http://www.gnu.org/software/gsl/gsl.html">www.gnu.org/software/gsl/gsl.html</A>. It
provides a vast amount of mathematical functions, of which we just use
a tiny bit.

<li> PyGSL allow us to call the GSL from Python. You can find the
sources at <A
HREF="http://pygsl.sourceforge.net/">pygsl.sourceforge.net</A>.
</ul></p>


<H4>Installation</H4> 

<p><ul>
<li> Download 
<A HREF="http://algorithmics.molgen.mpg.de/Downloads/GQL-Unix-1.0.zip">
GQL-Unix-1.0.zip</A>.
<li> Uncompress and untar the file you downloaded
<li> Set your PYTHON_PATH to include the directory where the ghmm.py has been
installed (typically {prefix}/lib/python2.X/site-packages)
<li> Note, your LD_LIBRARY_PATH must contain the directory where libghmm etc.
reside
</ul></P>


<H4>Troubleshooting</H4> 

<p><ul>
<li> Start the Python interpreter by typing 'python' in your shell. The following 
      commands should not produce no output.
      <ul>
	<li> import Tkinter
	<li> t = Tkinter.Tk() # A window should pop-up
	<li> import pygsl
	<li> import ghmm
      </ul>
      If you see error messages, check whether you used the correct
      Python version for installing (pygsl should be in site-packages
      in the lib/python2.X directory), whether your PATH and your
      LD_LIBRARY_PATH variable include the directories which contain
      the files you are trying to use. Also, if you did not install
      stuff yourself, check the protections.

<li> Messages in the console: GQL still outputs diagnostics and warnings.
     This is still a feature

<li> Why does GQL not have feature X? Feel free to implement it.
</ul></p>

<H3>Mac OS</H3> 

<H4>Prerequisites</H4> 

<p>You will need a Mac OS 10.3 or latter version.</p>


<H4>Installation</H4> 

<p><ul>
<li> Download 
<A HREF="http://algorithmics.molgen.mpg.de/Downloads/GQL-0.7-MAC.zip">
GQL-MAC-1.0.zip</A>
and <A HREF="http://algorithmics.molgen.mpg.de/Downloads/GQLCluster-0.7-MAC.zip">
 GQLCluster-MAC-1.0.zip</A>
<li> Uncompress the file you downloaded and double click it!
</ul></p>

<H3>Windows</H3> 

<H4>Prerequisites</H4> 

<p>You will need Windows 98 or latter version.</P>


<H4>Installation</H4> 

<p><ul>
<li> Download 
<A HREF="http://algorithmics.molgen.mpg.de/Downloads/GQL-Win-1.0.zip">
GQL-Win-1.0zip</A>.
<li> Uncompress the file you downloaded and double click it at GQLCluster or GQLQuery!
</ul>

See readme.txt avaliable in the unix/linux version for a non binary installation on windows.</P>




<H3>Documentation</H3>

<p> The papers bellows describes the methods implemented in GQL.

<blockquote><b>A. Schliep, A. Schönhuth, C. Steinhoff.</b><em> Using Hidden Markov Models to
Analyze Gene Expression Time Course Data.</em> Proceedings of the ISMB
2003. Bioinformatics. 2003 Jul; 19 Suppl 1: I255-I263
</blockquote>

<blockquote> <b>A. Schliep, C. Steinhoff, A. Schönhuth.</b><em>Robust
inference of groups in gene expression time-courses using mixtures of
HMM.</em> Proceedings of the ISMB 2004. Bioinformatics, Aug 2004; 20 Suppl 1:
I283 - I289. </blockquote>

<blockquote> <b>A. Schliep, I. G. Costa, C. Steinhoff, A. Schönhuth.</b><em>
Analysing gene expression time-courses </em>, IEEE Transactions on Computational Biology 
and Bioinformatics, to appear. </blockquote>

<blockquote> <b>I. G. Costa, A. Schönhuth, A. Schliep.</b><em> The
Graphical Query Language: a tool for analysis of gene expression
time-courses </em>, Bioinformatics,  2005, 21(10):2544-2545. </blockquote>
</p>

<H4>File formats:</H4>

<p> Both tools supports GHMM file formats for input data and model
descriptions (see GHMM). It also reads input files in standard tab
separated files, as the ones used by most of gene expression analysis
tools. In this format, each line represents a gene and the columns the
measured time points. The first column holds the gene identifiers and
the second column any type of annotation of the genes. Missing values
should be decoded as either 'Nan' or by not placing any character at
the position. Sample files of all formats are provided in <a
href='examples/'> examples</a>. </p>


<p><pre><kbd>

YHR124W	 meiosis                                -0.377685	-0.427071	-0.479749	 0.175438
YGR072W	 mRNA decay, nonsense-mediated unknown  -0.067600	-0.664033	-0.412644	 0.090134
YGR145W	 unknown                       	         0.266238	-0.854138	-0.103595	 0.371387
YIR031C  allantoin utilization                  -0.017010	 0.650807	 0.461851	-0.146432
YJR010W	 methionine biosynthesis                      NaN	 0.847968	 0.078140	-0.137952
YMR172W	 osmotic stress response                -0.734039	-0.258823	-0.135069	 0.127290
YIR032C  ureidoglycolate hydrolase              -0.287924	 0.701009	 0.464117	-0.160077
YHR053C	 metallothionein                        -0.263116	 0.780098	-0.363840	-0.396216
</kbd></pre></p>

<p>Example of a gene expression file during 4 time points. The second column holds
functional annotation of the genes.</p>

<p> GQL also use tab separated files for files containing partial labels. Now, 
the files have only two colunms, the first containing the gene id and
the second containing a numerical label (from 1 to n). </p>

<p><pre><kbd>

YHR124W  1
YGR072W  1
YGR145W  1
YIR031C  2 
YJR010W	 2 
YMR172W	 2 
YIR032C  2
YHR053C	 2
</kbd></pre></p>



<H4>Release Notes</H4> 

<p><b>Version 1.0:</b><br> 
We had this version in heavy use for
the last months. There are still some missing feature, and bugs.
<ul>

<li>Way too many log messages written to terminal


<li>GO Especificity X Entropy Threshold not yet included in the GUI.

<li>Priors on parameters needed to avoid overfitting (Just limit
the number of Baum-Welch steps for the moment)

<li>More options needed for estimation; update clustering view
periodically (right now, do a one-step computation, view the result
and estimate again)

<li>k-Means still not integrated

</ul></p>

</div>

<div id="Menu">
        <a href="index.html" title="">GHMM</a><br />
        <a href="installation.html" title="">Installation</a><br />
        <a href="documentation.html" title="">Documentation</a><br />
        <a href="hmmed.html" title="">HMMEd</a><br />
        <a href="gql.html" title="">GQL</a><br />
        <a href="contributors.html" title="">Contributors</a><br />
</div>

</body>
</html>
