<html>
<head>
<title>GHMM: General Hidden Markov Model library</title>
</head>
<style type="text/css" media="all">
@import "layout1.css" screen;
@import "print.css" print;
</style>
<div id="Content">

<p>
<img src="ghmm.gif">
</p>


<p>The General Hidden Markov Model library (GHMM) is a freely
available C library implementing efficient data structures and
algorithms for basic and extended HMMs with discrete and continous
emissions. It comes with Python wrappers which provide a 
much nicer interface and added functionality. The GHMM is
licensed under the LGPL.</P>

<center>
<img src="HMMEd-Montage.png">
</center>

<p>Defining a HMM with two states which output either heads or tails (think
of two coins which get exchanged occasionally) is
as easy as this:
<pre><kbd>
> python
Python 2.6.4 (r264:75706, Mar 16 2010, 09:46:46) 
[GCC 4.0.1 (Apple Inc. build 5490)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import ghmm
>>> sigma = ghmm.Alphabet(['h','t'])
>>> m = HMMFromMatrices(sigma, ghmm.DiscreteDistribution(sigma),
... [[0.9, 0.1], [0.3, 0.7]], [[0.5, 0.5], [0.2, 0.8]], [1.0, 0.0]) 
>>> print m
...
>>> help(m)
...
</kbd></pre>
Here the last three arguments for <kbd>HMMFromMatrices</kbd> are the
transition matrix, the emission matrix, and the initial
distribution. For example <kbd>0.9</kbd> is the probability of staying
in the first state. The first state uniformly emits heads or tails,
the second state produces tails with a probability of <kbd>0.8</kbd>.
The HMM always starts in the first state.
</p>


<h4>Features</h4>
<p>
<ul> 
<li>Discrete and continous emissions
<li>Mixtures of PDFs for continous emissions
<li>Non-homogenous Markov chains
<li>Pair HMMs
<li>Clustering and mixture modelling for HMMs
<li>Graphical Editor <a HREF="hmmed.html">HMMEd</a>
<li>Python bindings
<li>XML-based file format
<li>Portable (autoconf, automake)
</ul>
</p>


<h4>Development</h4> <p>The GHMM is under active development by the <A
HREF="http://bioinformatics.rutgers.edu">Alexander Schliep's group for
bioinformatics</A> at Rutgers University. The development is hosted at
Sourceforge <a
HREF="http://sourceforge.net/projects/ghmm/">http://sourceforge.net/projects/ghmm/</a>,
where you have access to the Subversion repository, mailing lists and
forums.</p>


<h4>Publications</h4> <p>The GHMM has been used in numerous scientific
publications by the core group. A GHMM software publication is forthcoming.
The GHMM has also extensively used as a teaching tool in Bioinformatics
and machine learning classes at the Freie Universit&auml;t Berlin and at
Rutgers.

<blockquote><b>A.&nbsp;Schönhuth, I.G.&nbsp;Costa and
A.&nbsp;Schliep.</b> Semi-supervised Clustering of Yeast Gene
Expression. In <em>Cooperation in Classification and Data
Analysis</em>, <em>Springer</em>, 151&ndash;160, 2009.
</blockquote>


<blockquote><b>Michael Seifert</b> <em>Analyzing Microarray Data Using
Homogenous and Inhomogenous Hidden Markov Models.</em> Diplomarbeit im
Studiengang Bioinformatik, Martin-Luther-Universit&auml;t Halle
(2006)</blockquote>


<blockquote><b>A. Schliep, I. G.  Costa, C. Steinhoff,
A. A. Sch&ouml;nhuth.</b> <em>Analyzing Gene Expression
Time-Courses</em> IEEE/ACM Transactions on Computational Biology and
Bioinformatics ( Special Issue on Machine Learning for
Bioinformatics), 2005, 2(3):179-193.
</blockquote>

<blockquote> <b>Matthias Heinig</b>
<em> Development of a Pair HMM based Gene Finder for the Paramecium Genome.</em>
Master Thesis, FU Berlin (2005)
</blockquote>

<blockquote> <b>I. G. Costa, A. Sch&ouml;nhuth, A. Schliep.</b><em> The
Graphical Query Language: a tool for analysis of gene expression
time-courses </em>, Bioinformatics,  2005, 21(10):2544-2545. 
</blockquote>

<blockquote> <b>A. Schliep, C. Steinhoff, A. Schönhuth.</b><em>Robust
inference of groups in gene expression time-courses using mixtures of
HMM.</em> Proceedings of the ISMB 2004.</blockquote>

<blockquote><b>A. Schliep, B. Georgi, W. Rungsarityotin, I. G. Costa,
A. Sch&ouml;nhuth</b> <em>The General Hidden Markov Model Library:
Analyzing Systems with Unobservable States </em>, Proceedings of the
Heinz-Billing-Price 2004: 121-136,.
</blockquote>

<blockquote> <b>A. Schliep, A. Schönhuth, C. Steinhoff.</b><em> Using
Hidden Markov Models to Analyze Gene Expression Time Course Data.</em>
Proceedings of the ISMB 2003. Bioinformatics. 2003 Jul; 19 Suppl 1:
I255-I263 </blockquote>

<blockquote><b>B.&nbsp;Knab, A.&nbsp;Schliep, B.&nbsp;Steckemetz and
B.&nbsp;Wichern</b>.  Model-Based Clustering With Hidden Markov Models
and its Application to Financial Time-Series Data. In <em>Between Data
Science and Applied Data Analysis</em>, <em>Springer</em>,
561&ndash;569, 2003.</blockquote>

<blockquote> <b>B. Georgi.</b> <em>A Graph-based Apporach to Clustering of
 Profile Hidden Markov Models</em>
Bachelor Thesis, FU Berlin. </blockquote>

<blockquote> <b>A. Weisse.</b> <em>Detecting Circular Permutations in
Remote Homologue Proteins.</em> Bachelor Thesis, FU Berlin.</blockquote>

<blockquote><b>Bernd Wichern.</b> <em>Hidden Markov Models for the
analysis of data from saving and loan banks</em> Ph.D. Thesis. ZAIK,
University of Cologne, Germany (2001). In German.
</blockquote>

<blockquote><b>Bernhard Knab.</b> <em>Extension of Hidden Markov
Models for the analysis of financial time-series data</em>
Ph.D. Thesis. ZAIK, University of Cologne, Germany (2000). In German.
</blockquote>
</p>




</div>

<div id="Menu">
        <a href="index.html" title="">GHMM</a><br />
        <a href="installation.html" title="">Installation</a><br />
        <a href="documentation.html" title="">Documentation</a><br />
        <a href="hmmed.html" title="">HMMEd</a><br />
        <a href="gql.html" title="">GQL</a><br />
        <a href="contributors.html" title="">Contributors</a><br />
</div>

</body>
</html>
